### PR TITLE
Fix scheduler hpo

### PR DESCRIPTION
**Pull request by razmikmelikbekyan**
*Issue #329 , if available:*

*Description of changes:*
Continuation of #345 for hyperband and rl.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/scheduler/hyperband.py [hunk 1]
```diff
@@ -219,6 +219,10 @@ def _run_reporter(self, task, task_job, reporter, searcher, terminator):
         last_updated = None
         while not task_job.done():
             reported_result = reporter.fetch()
+            if 'traceback' in reported_result:
+                logger.exception(reported_result['traceback'])
```
Comment (mseeger) [inline autogluon/scheduler/hyperband.py]: Please insert
   terminator.on_task_remove(task)
here. This is necessary for the terminator to keep its datastructures in order.

File: autogluon/scheduler/hyperband.py [hunk 2]
```diff
@@ -219,6 +219,10 @@ def _run_reporter(self, task, task_job, reporter, searcher, terminator):
...
+                break
```
Comment (mseeger) [inline autogluon/scheduler/hyperband.py]: Looking at this, and also FIFOScheduler:
Don't we also need to call
   reporter.move_on()  ?
I am not deep in the details here, but I note this is done if the reporter signals 'done', so should it not also be done if the reporter signals an exception?

File: autogluon/scheduler/rl_scheduler.py [hunk 3]
```diff
@@ -236,6 +236,11 @@ def _run_reporter(task, task_job, reporter):
             last_result = None
             config = task.args['config']
             while not task_job.done():
+                reported_result = reporter.fetch()
```
Comment (zhanghang1989) [inline autogluon/scheduler/rl_scheduler.py]: Why calling `reported_result = reporter.fetch()` twice?

File: autogluon/scheduler/hyperband.py (hunk 2 above)
Comment (razmikmelikbekyan) [reply autogluon/scheduler/hyperband.py]: yep, this is good catch, I added it.

File: autogluon/scheduler/hyperband.py (hunk 1 above)
Comment (razmikmelikbekyan) [reply autogluon/scheduler/hyperband.py]: done

**Comment by szha** [comment]
Job PR-354-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-354/2/index.html

**Comment by szha** [comment]
Job PR-354-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-354/3/index.html

**Comment by Innixma** [comment]
@mseeger You're review still shows as requesting changes, can you change to approved so that the PR can be merged?

**Comment by Innixma** [comment]
Merging into mainline, thanks again for your contribution!

**Pull request merged**